### PR TITLE
Pr/membership

### PR DESCRIPTION
**Pull request by mhewson**
## Description
membership palettes

### CODE REVIEW

**Review by c4rlosviteri — APPROVED** [review]

File: src/styles/main.scss [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /* Tokens */
 @import '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/scss/cdr-tokens';
+@import '/node_modules/@rei/cdr-tokens/dist/rei-dot-com/css/cdr-tokens';
```
Comment (sikhote) [inline src/styles/main.scss]: Do we need both of these? (line 2 & 3)

File: src/components/surface/CdrSurface.vue [hunk 2]
```diff
@@ -18,20 +18,25 @@ const props = withDefaults(defineProps<surface>(), {
...
+    class: mapClasses(style, ...classes) || undefined,
+  };
 });
 </script>
 
 <template>
   <component
     :is="tag"
     v-bind="rootProps"
+    :data-palette="props.palette"
```
Comment (sikhote) [inline src/components/surface/CdrSurface.vue]: I think this is better added to the `rootProps` return, something like 

```
  return {
    class: mapClasses(style, ...classes) || undefined,
    'data-palette': props.palette,
  };
```

File: src/components/surface/examples/Surface.vue [hunk 3]
```diff
@@ -36,20 +36,15 @@ const boxes: Example[] = [
   {
     label: 'This surface uses the primary modifier to apply a background and border color',
```
Comment (sikhote) [inline src/components/surface/examples/Surface.vue]: I suggest this label is updated to reflect the example, maybe `Showing the use of `membership-subtle` palette

Comment (sikhote) [reply src/components/surface/examples/Surface.vue]: Same with the label below as well

File: src/dev/SinkWrapper.vue [hunk 4]
```diff
@@ -37,7 +37,7 @@ export default {
   data() {
     return {
       palette: this.$route.query.palette || 'default',
-      palettes: ['default', 'sandstone'],
+      palettes: ['default', 'sandstone', 'membership-vibrant', 'membership-subtle'],
```
Comment (sikhote) [inline src/dev/SinkWrapper.vue:40]: 👍

File: src/styles/cdr-reset.scss [hunk 5]
```diff
@@ -114,9 +114,4 @@ input[type='month'] {
   -webkit-appearance: none;
 }
 
-[data-palette='sandstone'] {
-  --cdr-color-background-surface-primary: #{$cdr-color-background-secondary};
-  --cdr-color-background-surface-secondary: #dcd6cb;
-  --cdr-color-border-surface-primary: #{$cdr-color-border-secondary};
-  --cdr-color-border-surface-secondary: #726d64;
-}
+
```
Comment (sikhote) [inline src/styles/cdr-reset.scss]: With this removal, how are users expected to add in palettes? Maybe they would need to import the palettes file manually?

**Comment by sikhote** [comment]
Overall looks good, just looking to clarify a few things

File: src/styles/settings/palette.vars.scss [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
+[data-palette='sandstone'] {
+  --cdr-color-background-surface-primary: #{$cdr-color-background-secondary};
+  --cdr-color-background-surface-secondary: #dcd6cb;
+  --cdr-color-border-surface-primary: #{$cdr-color-border-secondary};
+  --cdr-color-border-surface-secondary: #726d64;
+}
+
+[data-palette='membership-subtle'] {
+  --cdr-color-background-surface-primary: var(--cdr-membership-subtle-color-background-primary);
```
Comment (sikhote) [inline src/styles/settings/palette.vars.scss]: If these get changed to sass variables, solves the double import issue perhaps

File: src/components/surface/CdrSurface.vue [hunk 7]
```diff
@@ -9,28 +9,72 @@ import { getSurfaceProps } from '../../utils/surface';
 defineOptions({ name: 'CdrSurface' });
 
 const props = withDefaults(defineProps<surface>(), {
+    /**
```
Comment (sikhote) [inline src/components/surface/CdrSurface.vue]: FYI this type of documentation is also part of the `surface` interface type. I'm not sure which one the doc site will pick up. I could see it as a little confusing to have both for someone coming into it without background knowledge, but I don't mind.

File: src/styles/cdr-reset.scss (hunk 5 above)
Comment (sikhote) [reply src/styles/cdr-reset.scss]: I think this question still applies—should we export the palette sass file you created?

**Review by c4rlosviteri — APPROVED** [review]

File: src/components/surface/CdrSurface.vue (hunk 7 above)
Comment (mhewson) [reply src/components/surface/CdrSurface.vue]: Oh my bad - interface seems like the correct location for this - the doc site seems to be using slightly different language

File: src/styles/cdr-reset.scss (hunk 5 above)
Comment (mhewson) [reply src/styles/cdr-reset.scss]: I ended up moving this file to styles root dir and allowing it to be built to dist

File: src/styles/settings/palette.vars.scss (hunk 6 above)
Comment (mhewson) [reply src/styles/settings/palette.vars.scss]: yes I added sass as a fallback

**Review by sikhote — APPROVED** [review]
Sweeet ✅

**Pull request merged**